test/e2e/addon/podidentityverify.go

```diff
@@ -45,7 +50,49 @@ func (v VerifyPodIdentityAddon) Run(ctx context.Context) error {
 		return err
 	}
 
-	// TODO: Deploy a pod with service account to verify the pod identity token section is populated in pod manifest files
+	// Deploy a pod with service account to verify the pod identity token section is populated in pod yaml
+	node, err := kubernetes.WaitForNode(ctx, v.K8S, v.NodeIP, v.Logger)
+	if err != nil {
+		return err
+	}
+
+	nodeName := node.Name
+	podName := kubernetes.GetNginxPodName(nodeName)
+	v.Logger.Info("Creating a test pod on the hybrid node for pod identity add-on")
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "nginx",
+					Image: fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com/ecr-public/nginx/nginx:latest", constants.EcrAccounId, "us-west-2"),
+				},
+			},
+			// schedule the pod on the specific node using nodeSelector
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": nodeName,
+			},
+			ServiceAccountName: podIdentityServiceAccount,
+		},
+	}
+	if err = kubernetes.CreateNginxPod(ctx, v.K8S, pod, v.Logger); err != nil {
+		return err
+	}
+
+	// Get pod and verify the volume mount for pod identity add-on
+	// Ref: https://docs.aws.amazon.com/eks/latest/userguide/pod-id-how-it-works.html#pod-id-agent-pod
+	if pod, err := v.K8S.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{}); err != nil {
+		return err
+	} else {
+		for _, volumeMount := range pod.Spec.Containers[0].VolumeMounts {
+			if volumeMount.Name == podIdentityToken {
+				return nil
+			}
+		}
+	}
```

Comment:
This is clever. However, this is kind of testing the implementation instead of testing the feature
Why don't we just try to make use of the aws creds that should be assigned to the pod? That would be closer to testing the feature e2e. Regardless of how the creds are mounted on the pod.

Reply:
Yeah i agree, i was kind of thinking of actually use the aws cli in a container and do like an s3 ls or something, wahtever permission we want to give it?